protocol/react/README.md

```diff
@@ -0,0 +1,121 @@
+  component.
+* If our app needs to make network requests and hold state outside of a
+  component, we like using GraphQL and our preferred library is
+[Apollo](https://www.apollographql.com). We have a [section on
+`GraphQL`](https://github.com/thoughtbot/guides/blob/master/graphql/README.md)
+in our Guides.
+* When building React apps with TypeScript and Apollo, we've found working
+  in [VSCode](https://code.visualstudio.com/) to be a mostly-good experience.
+
+Set Up Laptop
+-------------
+
+Set up your laptop with [this script](https://github.com/thoughtbot/laptop) and
+[these dotfiles](https://github.com/thoughtbot/dotfiles).
+
+Create App
+----------
+
+In the terminal, initialize the app with
+[create-react-app](https://github.com/facebook/create-react-app) and TypeScript.
+`create-react-app` is an open source React boilerplate tool similar to
+Thoughtbot's own Suspenders for Rails. It creates a basic React app with no
+build configuration and a good set of defaults.
+
+    yarn create react-app app-name --typescript
+
+Set Up App
+----------
+
+We're huge fans of automatic code formatting and linting.
+[Prettier](https://prettier.io) and [ESLint](https://eslint.org/) are both great
+libraries to facilitate this.
+
+Add `eslint` to the project
+
+    yarn add eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin --dev
+
+The following is a `.eslintrc.js` file that has some sensible defaults. Add this
+to the root of your project.
+
+<details>
+
+    module.exports = {
+      env: {
+        node: true,
+        es6: true,
+      },
+      parser: '@typescript-eslint/parser',
+      plugins: [],
+      extends: [
+        'plugin:@typescript-eslint/recommended',
+        'prettier/@typescript-eslint',
+        'plugin:prettier/recommended',
+      ],
+      parserOptions: {
+        ecmaVersion: 2018,
+        sourceType: 'module',
+      },
+      rules: {
+        'no-console': 'error',
+        'require-await': 'error',
+        'no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
+        'no-var': 'error',
+        'max-len': [
+          'error',
+          120,
+          {
+            ignorePattern: '^import\\s.+\\sfrom\\s.+;$',
+          },
+        ],
+        '@typescript-eslint/consistent-type-assertions': 'off',
+        '@typescript-eslint/no-angle-bracket-type-assertion': 'off',
+        '@typescript-eslint/no-non-null-assertion': 'error',
+        '@typescript-eslint/no-explicit-any': 'error',
+        '@typescript-eslint/no-unused-vars': [
+          'error',
+          {
+            ignoreRestSiblings: true,
+            argsIgnorePattern: '^_',
+          },
+        ],
+      },
+    };
+</details>
+
+And here's a `.prettierrc.js` file with some good default code styles.
+
+<details>
+
+    module.exports =  {
+      semi:  true,
+      trailingComma:  'all',
+      singleQuote:  true,
+      printWidth:  120,
+      tabWidth:  2,
+    };
```

Comment:
Some of my previous notes on the prettier config seemed to be lost in the shuffle:

- `printWidth` ideally would be 80 as this matches our other guides, and [perttier's own recommendations](https://prettier.io/docs/en/options.html#print-width)
- `semi: true` is default, so we can omit
- `tabWidth: 2` is default, so we can omit